### ISSUE TITLE
sbt 1.4.0: BSP does not report about compilation errors resolved by deleting an offending file

### ISSUE THREAD

**Issue by ritschwumm** [sbt/sbt#5937]
## steps

* run `sbt ~compile` in an existing project
* connect to the bsp socket and print messages
* create a new scala file containing some invalid syntax
* see the errors being reported in BSP
* delete the bad scala file, thus resolving the compilation error

## problem

there are no PublishDiagnostics message in BSP for the deleted file, so the BSP client does not know the compilation error has been resolver 

## expectation

diagnostics are published in a way a client can remove compilation errors when they no longer exist

**Comment by adpi2** [comment]
@ritschwumm I spent some time on this issue but I have not yet found a good way to solve it. And I am not sure it is worth it.

I think it is ok that a BSP server does not send diagnostics on files that no longer exist, because diagnostics are sent during compilation based on the current list of sources.

**Comment by ritschwumm** [comment]
i agree it would be a bit weird to send diagnostics for a file that does not exists.

on the other hand: as the part of the client displaying compilation errors, i'm mostly interested in maintaining a complete list of currently relevant compilation errors (and warnings). the files themselves i don't care too much about. 

how about, instead of sending diagnostics, just telling the client when a file that was relevant in a previous compilation is now gone, so i can just remove all errors concerning this file? the only other alternative i could see is keep a list of files and then manually check (from time to time? when exactly?) which of these still exist.
